packages/server/src/lib/grants-ingest.js

```diff
@@ -58,6 +58,13 @@ function mapSourceDataToGrant(source) {
         grant.opportunity_status = 'archived';
     } else if (today.isSameOrAfter(moment(grant.close_date), 'date')) {
         grant.opportunity_status = 'closed';
+    } else if (today.isBefore(moment(grant.open_date), 'date')) {
+        // depending on how we model the estimated open date, we may need to do the 
+        // date comparison with the new field for estimated date
+        grant.opportunity_status = 'forecasted'
+        // Note: in the knex query, we are deriving the status based on 
+        // archive_date and close_date. Probably we just want to pass
+        // the opportunity_status to the frontend, and not re-derive it here?
```

Comment:
Unlike the transition from forecasted to non-forecasted, there's (currently) no explicit grants-ingest event that we expect to receive that would allow us to store a new materialized value for the grant being closed or archived. In other words, to determine whether a grant is forecasted or fully published, we have these options:
- Check `is_forecasted` / some value that signifies whether the upstream record was sourced only from forecasted data
- Check whether `post_date` is in the future

However, the `post_date` may well be the last time we ever see an event from grants-ingest for a particular grant, so the status of being `closed` or `archived` need to be derived from a check like
```js
if (close_date <= today) {
  if (archive_date > today) {
    status = 'closed';
  } else {
    status = 'archived';
  }
}
```

(Hopefully that's useful and I'm not misunderstanding the question!)